src/transformers/models/owlvit/modeling_tf_owlvit.py

```diff
@@ -0,0 +1,1730 @@
+
+        # Resize to [batch_size, num_patches, num_patches, hidden_size]
+        new_size = (
+            image_embeds.shape[0],
+            int(np.sqrt(image_embeds.shape[1])),
+            int(np.sqrt(image_embeds.shape[1])),
+            image_embeds.shape[-1],
+        )
+        image_embeds = tf.reshape(image_embeds, new_size)
+
+        return (image_embeds, text_embeds)
+
+    @unpack_inputs
+    @add_start_docstrings_to_model_forward(OWLVIT_OBJECT_DETECTION_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=TFOwlViTObjectDetectionOutput, config_class=OwlViTConfig)
+    def call(
+        self,
+        input_ids: Optional[TFModelInputType] = None,
+        pixel_values: Optional[TFModelInputType] = None,
+        attention_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> TFOwlViTObjectDetectionOutput:
+        r"""
+        Returns:
+
+        Examples:
+        ```python
+        >>> import requests
+        >>> from PIL import Image
+        >>> import tensorflow as tf
+        >>> from transformers import OwlViTProcessor, TFOwlViTForObjectDetection
+
+        >>> processor = OwlViTProcessor.from_pretrained("google/owlvit-base-patch32")
+        >>> model = OwlViTForObjectDetection.from_pretrained("google/owlvit-base-patch32")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> texts = [["a photo of a cat", "a photo of a dog"]]
+        >>> inputs = processor(text=texts, images=image, return_tensors="tf")
+        >>> outputs = model(**inputs)
+        ```"""
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+
+        # Return last hidden states of text and vision transformers
+        text_model_last_hidden_states = None
+        vision_model_last_hidden_states = None
+
+        if output_hidden_states:
+            outputs = self.owlvit(
+                input_ids=input_ids,
+                pixel_values=pixel_values,
+                attention_mask=attention_mask,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+            )
+
+            text_model_last_hidden_states = outputs[-2][0]
+            vision_model_last_hidden_states = outputs[-1][0]
+
+        # Embed images and text queries
+        feature_map, query_embeds = self.image_text_embedder(
+            input_ids=input_ids,
+            pixel_values=pixel_values,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+        )
+
+        batch_size, height, width, hidden_dim = feature_map.shape
```

Comment:
```suggestion
        batch_size, height, width, hidden_dim = tf.shape(feature_map)
```

Reply:
This should have been:
`batch_size, num_patches, num_patches, hidden_dim = feature_map.shape`

Fixing this and all relevant parts.